src/types/json.h

```diff
@@ -65,6 +66,47 @@ struct JsonValue {
     return Status::OK();
   }
 
+  Status StrAppend(std::string_view path, const std::string &append_value, std::vector<uint64_t> &append_cnt) {
+    try {
+      auto append_trimmed = util::Trim(append_value, "\"");
+      jsoncons::jsonpath::json_replace(
+          value, path, [&append_trimmed, &append_cnt](const std::string & /*path*/, jsoncons::json &origin) {
+            if (origin.is_string()) {
+              auto origin_str = origin.to_string();
+              // need trim "
+              auto origin_trimmed = util::Trim(origin_str, "\"");
+              append_cnt.push_back(origin_trimmed.length() + append_trimmed.length());
+              std::string new_value;
+              new_value.append("\"").append(origin_trimmed).append(append_trimmed).append("\"");
+              origin = jsoncons::json::parse(new_value);
+            } else {
+              append_cnt.push_back(0);
+            }
+          });
+    } catch (const jsoncons::jsonpath::jsonpath_error &e) {
+      return {Status::NotOK, e.what()};
+    }
+
+    return Status::OK();
+  }
+
+  Status StrLen(std::string_view path, std::vector<uint64_t> &str_lens) const {
+    try {
+      jsoncons::jsonpath::json_query(value, path,
+                                     [&str_lens](const std::string & /*path*/, const jsoncons::json &origin) {
+                                       if (origin.is_string()) {
+                                         // need subtraction "
+                                         str_lens.push_back(origin.to_string().length() - 2);
```

Comment:
IMHO even you trim these two quotation marks here, I think the result still can be incorrect. e.g. `"\n"` -> `2`

Reply:
ohhh,  should be use `as<std::string>` or `as_string`, it will automatically process `"`